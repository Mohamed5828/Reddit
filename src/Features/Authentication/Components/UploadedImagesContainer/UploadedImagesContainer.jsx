// Import components
import UploadedImage from "Features/Authentication/Components/UploadedImage/UploadedImage";

// Import styled components
import { ThumbsContainer } from "../UploadedImagesContainer/UploadedImagesContainer.styled";

/**
 * Container for uploaded images
 *
 * @param {Array} files - Array of files (state)
 * @param {Function} setFiles - Function to set files (state)
 * @param {Function} open - Function to open file explorer
 * @param {Number} selectedImageId - Id of selected image (state)
 * @param {Function} setSelectedImageId - Function to set selected image id (state)
 *
 * @returns {React.Component} - UploadedImagesContainer component
 */
const UploadedImagesContainer = ({ file, setFile, open }) => {
  /**
   * Function to handle delete file
   *
   * @param {File} file - The file to be removed
   */
  const deleteFile = (file) => {
    setFile(null); // update the state
  };
  return (
    <ThumbsContainer>
      <UploadedImage
        key={file.name + file.uploadDate}
        id={file.name + file.uploadDate}
        file={file}
        deleteFile={deleteFile}
        open={open}
      />
    </ThumbsContainer>
  );
};

export default UploadedImagesContainer;
