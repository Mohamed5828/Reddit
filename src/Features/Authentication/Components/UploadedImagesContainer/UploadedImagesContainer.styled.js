// Import styled
import styled from "styled-components";

export const ThumbsContainer = styled.aside`
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  margin-top: 16;
  overflow: auto;
`;

export const UploadIcon = styled.button`
  width: 100px;
  min-width: 100px;
  height: 100px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0 12px 12px 0;
  display: inline-block;
  color: ${({ theme }) => theme.color.muted};
  border-radius: 4px;
  background-color: ${({ theme }) => theme.background.primary};
  border: 1px dashed ${({ theme }) => theme.lineColor.primary};
`;
