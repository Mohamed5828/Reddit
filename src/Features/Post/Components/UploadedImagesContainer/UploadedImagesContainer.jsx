// Import icons
import { useState } from "react";
import { BsPlusLg } from "react-icons/bs";

// Import components
import UploadedImage from "Features/Post/Components/UploadedImage/UploadedImage";

// Import styled components
import {
  ThumbsContainer,
  UploadIcon,
} from "../UploadedImagesContainer/UploadedImagesContainer.styled";

/**
 * Container for uploaded images
 *
 * @param {Array} files - Array of files (state)
 * @param {Function} setFiles - Function to set files (state)
 * @param {Function} open - Function to open file explorer
 * @param {Number} selectedImageId - Id of selected image (state)
 * @param {Function} setSelectedImageId - Function to set selected image id (state)
 *
 * @returns {React.Component} - UploadedImagesContainer component
 */
const UploadedImagesContainer = ({
  files,
  setFiles,
  open,
  selectedImageId,
  setSelectedImageId,
}) => {
  // State to state if the image is selected
  const [isImageSelected, setIsImageSelected] = useState(false);

  /**
   * Function to handle the click on the image
   *
   * @param {String} id The id of the selected image
   */
  const handleImageClick = (id) => {
    // Set the isSelectedImage boolean to true to show that there is an image selected
    setIsImageSelected(true);

    // Set the selected image id
    setSelectedImageId(id);
  };

  /**
   * Function to handle delete file
   *
   * @param {File} file - The file to be removed
   */
  const deleteFile = (file) => {
    const newFiles = [...files]; // make a var for the new array
    newFiles.splice(newFiles.indexOf(file), 1); // remove the file from the array
    setFiles(() => newFiles); // update the state
  };
  return (
    <ThumbsContainer>
      {files.map((file) => (
        <UploadedImage
          key={file.name + file.uploadDate}
          id={file.name + file.uploadDate}
          file={file}
          deleteFile={deleteFile}
          isSelected={isImageSelected}
          selectedImageId={selectedImageId}
          handleClick={handleImageClick}
        />
      ))}
      {files.length !== 0 && (
        <UploadIcon variant="light" onClick={open}>
          <BsPlusLg />
        </UploadIcon>
      )}
    </ThumbsContainer>
  );
};

export default UploadedImagesContainer;
