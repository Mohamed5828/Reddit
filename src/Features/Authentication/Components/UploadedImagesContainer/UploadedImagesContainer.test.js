import { shallow } from "enzyme";
import React from "react";
import { useState } from "react";

// Import components
import UploadedImagesContainer from "./UploadedImagesContainer";

describe("Uploaded images container", () => {
  let file = { name: "image", uploadDate: "2020-10-10" };
  const setFile = (recFile) => (file = recFile);
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <UploadedImagesContainer file={file} setFile={setFile} />
    );
  });

  it("should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
