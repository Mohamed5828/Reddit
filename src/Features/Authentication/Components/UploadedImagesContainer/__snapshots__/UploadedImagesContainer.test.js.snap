// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uploaded images container should render without crashing 1`] = `
<styled.aside>
  <UploadedImage
    deleteFile={[Function]}
    file={
      Object {
        "name": "image",
        "uploadDate": "2020-10-10",
      }
    }
    id="image2020-10-10"
    key="image2020-10-10"
  />
</styled.aside>
`;
